src/NinjaTools.FluentMockServer.API/Proxy/Visitors/ComparasionVisitor.cs

```diff
@@ -0,0 +1,171 @@
+                return Pass();
+
+            Guard.Against.NullOrEmpty(visitable.Header, nameof(visitable) + nameof(Headers));
+
+            if (HttpRequest.Headers is null && visitable.Header.EnsureNotNullNotEmpty() is { })
+                return Fail();
+
+            var success = visitable.Header.Where(v =>
+            {
+                if (!HttpContext.Request.Headers.TryGetValue(v.Key, out var value))
+                    return false;
+
+                return !value.Except(v.Value).Any();
+
+            }).Any();
+
+            return success ? Pass() : Fail();
+        }
+
+
+        /// <inheritdoc />
+        public double Visit(Cookies? visitable, CancellationToken token = default)
+        {
+            token.ThrowIfCancellationRequested();
+            if (!visitable?.Any() ?? true)
+                return Pass();
+
+            Guard.Against.NullOrEmpty(visitable.Cookie, nameof(visitable) + nameof(Cookies));
+
+            if (HttpRequest.Cookies is null && visitable.Cookie.EnsureNotNullNotEmpty().Any())
+                Fail();
+
+            if (visitable.Cookie.Except(HttpContext.Request.Cookies).EnsureNotNullNotEmpty().Any())
+                return Fail();
+
+            return Pass();
+        }
+
+        /// <inheritdoc />
+        public double Visit(Query visitable, CancellationToken token = default)
+        {
+            token.ThrowIfCancellationRequested();
+            if (!visitable?.IsEnabled() ?? true)
+                return Pass();
+
+            Guard.Against.NullOrEmpty(visitable.QueryString.Value, nameof(visitable) + nameof(Query));
+            if (!HttpRequest.QueryString.HasValue && visitable.QueryString.HasValue)
+                return Fail();
+
+            if (visitable.QueryString.Value == HttpRequest.QueryString.Value)
+                return Pass();
+
+            return Fail();
+        }
+
+        /// <inheritdoc />
+        public double Visit(Path? visitable, CancellationToken token = default)
+        {
+            token.ThrowIfCancellationRequested();
+            if (!visitable?.IsEnabled() ?? true)
+                return Pass();
+
+            Guard.Against.NullOrWhiteSpace(visitable, nameof(visitable) + nameof(Path));
+
+            if (!HttpRequest.Path.HasValue && visitable is { } v && v.PathString.HasValue)
+                return  Fail();
+
+            if (visitable.ToPath() == HttpRequest.Path.Value)
+                return Pass();
+
+            return  Fail();
+        }
+
+        /// <inheritdoc />
+        public double Visit(Method? visitable, CancellationToken token = default)
+        {
+            token.ThrowIfCancellationRequested();
+            if (!visitable?.IsEnabled() ?? true)
+                return Pass();
+
+            Guard.Against.NullOrEmpty(visitable, nameof(visitable) + nameof(Method));
+            if (string.IsNullOrWhiteSpace(HttpRequest.Method) && visitable is { } v && !string.IsNullOrWhiteSpace(v.MethodString))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/git_36/NinjaTools.FluentMockServer/pullRequest?prid=4802644)